### PR TITLE
update partition-table.md

### PR DESCRIPTION
**Pull request by gengliqi**
Signed-off-by: Liqi Geng <gengliqiii@gmail.com>

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3216
- Other reference link(s):

### CODE REVIEW

File: partitioned-table.md [hunk 1]
```diff
@@ -11,13 +11,13 @@ This document introduces TiDB's implementation of partitioning.
...
 
-This section introduces the types of partitioning which are available in TiDB. Currently, TiDB supports range partitioning and hash partitioning.
+This section introduces the types of partitioning which are available in TiDB. Currently, TiDB supports Range partitioning and Hash partitioning.
 
 Range partitioning is used to resolve the performance issues caused by a large amount of deletions in the application, and it supports fast drop partition operations. Hash partitioning is used to scatter the data when there are a large amount of writes.
 
 ### Range partitioning
 
-When a table is partitioned by range, each partition contains rows for which the partitioning expression value lies within a given range. Ranges have to be contiguous but not overlapping. You can define it by using the `VALUES LESS THAN` operation.
+When a table is partitioned by Range, each partition contains rows for which the partitioning expression value lies within a given Range. Ranges have to be contiguous but not overlapping. You can define it by using the `VALUES LESS THAN`.
```
Comment (TomShawn) [inline partitioned-table.md:20]:
```suggestion
When a table is partitioned by Range, each partition contains rows for which the partitioning expression value lies within a given Range. Ranges have to be contiguous but not overlapping. You can define it by using `VALUES LESS THAN`.
```

File: partitioned-table.md [hunk 2]
```diff
@@ -385,6 +385,11 @@ Empty set (0.00 sec)
 
 You can see that the inserted record `(NULL, 'mothra')` falls into the same partition as `(0, 'gigan')`.
 
+> **Attention**
```
Comment (TomShawn) [inline partitioned-table.md:388]:
```suggestion
> **Note:**
```

File: partitioned-table.md [hunk 3]
```diff
@@ -385,6 +385,11 @@ Empty set (0.00 sec)
...
+> The processing of NULL by Hash partition here is the same as [Document description of MySQL](https://dev.mysql.com/doc/refman/8.0/en/partitioning-handling-nulls.html), but it is not consistent with the actual behavior of MySQL. In other words, MySQL's documentation is not consistent with its implementation.
```
Comment (TomShawn) [inline partitioned-table.md:389]:
```suggestion
> `NULL` values by Hash partitions in TiDB are handled in the same way as described in [How MySQL Partitioning Handles NULL](https://dev.mysql.com/doc/refman/8.0/en/partitioning-handling-nulls.html), which, however, is not consistent with the actual behavior of MySQL. In other words, MySQL's implementation in this case is not consistent with its documentation.
```

File: partitioned-table.md [hunk 4]
```diff
@@ -385,6 +385,11 @@ Empty set (0.00 sec)
...
+> 
+> The final behavior of TiDB is subject to the description in this document.
```
Comment (TomShawn) [inline partitioned-table.md:391]:
```suggestion
> In this case, the actual behavior of TiDB is in line with the description of this document.
```

File: partitioned-table.md [hunk 5]
```diff
@@ -886,20 +901,37 @@ Query OK, 0 rows affected (0.12 sec)
 
 You can add a non-unique index by using `ALTER TABLE` statements. But if you want to add a unique index, the `c1` column must be included in the unique index.
 
+When using a partitioned table, the prefix index cannot be specified as a unique attribute:
```
Comment (TomShawn) [inline partitioned-table.md:904]:
```suggestion
When using a partitioned table, you cannot specify the prefix index as a unique attribute:
```

File: partitioned-table.md [hunk 6]
```diff
@@ -1017,3 +1049,7 @@ select * from t;
 +------|------+
 5 rows in set (0.00 sec)
 ```
+
+The environment variable `tidb_enable_table_partition` can control whether to enable the partition table function. If this variable is set to `off`, the partition information will be ignored when the table is built, and the table will be built as a normal table.
```
Comment (TomShawn) [inline partitioned-table.md:1053]:
```suggestion
The `tidb_enable_table_partition` environment variable controls whether to enable the partitioned table feature. If this variable is set to `off`, the partition information will be ignored when a table is created, and this table will be created as a normal table.
```

File: partitioned-table.md [hunk 7]
```diff
@@ -1017,3 +1049,7 @@ select * from t;
...
+
+This variable is only used to build the table, and it is invalid to modify the variable after the table has been built. For details, see [TiDB specific system variables] (/tidb-specific-system-variables.md#tidb_enable_table_partition).
```
Comment (TomShawn) [inline partitioned-table.md:1055]:
```suggestion
This variable is only used in table creation. After the table is created, it is invalid to modify this variable value. For details, see [TiDB specific system variables](/tidb-specific-system-variables.md#tidb_enable_table_partition).
```

File: partitioned-table.md [hunk 8]
```diff
@@ -1017,3 +1049,7 @@ select * from t;
 +------|------+
 5 rows in set (0.00 sec)
 ```
+
+The `tidb_enable_table_partition` environment variable controls whether to enable the partitioned table feature. If this variable is set to `off`, the partition information will be ignored when a table is created, and this table will be created as a normal table.
+
+This variable is only used in table creation. After the table is created, it is invalid to modify this variable value. For details, see [TiDB specific system variables](/tidb-specific-system-variables.md#tidb_enable_table_partition).
```
Comment (tiancaiamao) [inline partitioned-table.md:1055]:
```suggestion
This variable is only used in table creation. After the table is created, modify this variable value takes no effect. For details, see [TiDB specific system variables](/tidb-specific-system-variables.md#tidb_enable_table_partition).
```

Comment (tiancaiamao) [reply partitioned-table.md:1055]: Rest LGTM

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**